src/inventory.gql

```diff
@@ -16,12 +16,44 @@ type ProductItem {
 """A simple UUID parser"""
 scalar UUID
 
+type PaginatedProductItem {
+  nodes: [ProductItem!]
+  totalCount: Int!
+  hasNextPage: Boolean!
+}
+
 type Query {
-  productItems: [ProductItem!]!
+  productItems(findProductItemsInput: FindProductItemArgs!): PaginatedProductItem!
   countProductItems(productVersionId: UUID!): Int!
```

Comment:
in the future this should also be done via a connection, you might decide to do this in the federation PR, I just want to make sure your connection infrastructure supports multiple connections

Reply:
So you are proposing to return all _ProductItems_ with a given _productVersionId_ instead of a count?
Propably this also involves the "return as much as possible" pattern, but this query was more directed to the fronend to get the stock count.

Reply:
no, I want a connection on ProductVariant which allows you to both
- get the total count of inventory items associated with that ProductVariant 
- get the associated inventory items paginated (as usual with connections)

Reply:
Sure I could do that, but I think that this will lead to (avoidable) performance issues.

Since the query would be used rather often and its purpose is just to indicate the stock count.
I cant image a use-case, where the consumer would care about the actual inventory productItem Id.

The reserve part will be a different mutation.
I was thinking about adding a seperate collection with just the stock count for each productVariant to boost performance. Returning all productItems for the count query would prohibit that.

Reply:
as we discussed, when totalCount / nodes are resolved independently, this should not lead to performance issues